dame_flame/flame_algorithm.py

```diff
@@ -89,13 +90,21 @@ def flame_generic(df_all, treatment_column_name, outcome_column_name,
                   
     return_matches = pd.DataFrame(columns=all_covs, index=df_all.index)
     # As an initial step, we attempt to match on all covariates
-    
+
     covs_match_on = all_covs
-    
-    matched_rows, return_matches = grouped_mr.algo2_GroupedMR(
+    matched_rows, return_matches, units_in_g = grouped_mr.algo2_GroupedMR(
         df_all, df_unmatched, covs_match_on, all_covs, treatment_column_name, 
         outcome_column_name, return_matches)
-
+# =============================================================================
```

Comment:
Clean up formatting yes, but can you also put a one line comment in, detailing what the point of this loop is in plain language? ie I have stuff like "Now remove the matched units", "here we begin the iterative dropping procedure"...